enhancements/testing/openshift-tests-extension.md

```diff
@@ -0,0 +1,1202 @@
+                # is a sub-suite of openshift/conformance. All information
+                # about a suite is additive across extension binaries.
+                # For example, multiple binaries can advertise the same
+                # suite -- if they advertise different parents, then
+                # origin will treat the suite as a subset of each 
+                # identified parent.
+                "openshift/conformance"
+            ],
+
+            # Test cases can be advertised from across a number
+            # of extension binaries. The following cel expressions
+            # are OR'd together. If they select a test advertised
+            # from another binary, they will be considered part of
+            # this advertised suite.
+            # Test tags, names, and other attributes will be
+            # described later.
+            "qualifiers": [
+                "(test.tags.suite==\"fips\" || test.name.contains(\"fips\")) && !test.labels.has(\"Disruptive\")",
+                # Normally, CEL expressions evaluate against ALL test
+                # specs discovered by origin, regardless of the extension
+                # that advertised them. This makes it easy for 
+                # a suite defined in one extension to "absorb" tests 
+                # specified by another extension. To limit the expression
+                # to tests in only certain extensions, you can filter on "source"
+                "source = \"openshift:payload:hyperkube\" && test.name.contains(\"FIPS\"))"
+            ]
+        }
+    ],
+}
+```
+
+##### List Tests - Extension Test Listing
+
+The "list tests" action will output information about tests exposed by the extension
+for different environments. A test "environment" describes the attributes of
+cluster under test (e.g. the cloud provider, the topology, etc).
+
+The implementation should eliminate tests that are not appropriate for the
+specified environment.
+
+The arguments accepted by the verb may vary over time and will be
+defined by the extension interface version implemented by the binary.
+
+**Version 1**
+
+```
+$ extension-binary list tests 
+Environment Information:
+  --component     "default" or "<product>:<type>:<component>"
+  --platform      The hardware or cloud platform ("aws", "gcp", "metal", ...).
+  --network       The network of the target cluster ("ovn", "sdn"). 
+  --upgrade       The upgrade that was performed prior to the test run ("micro", "minor").
+  --topology      The target cluster topology ("ha", "microshift", ...).
+  --architecture  The CPU architecture of the target cluster ("amd64", "arm64").
+  --installer     The installer used to create the cluster ("ipi", "upi", "assisted", ...).
+  --config        Multiple. Non-default component configuration in the environment.
+  --fact          Multiple. Facts advertised by cluster components.
+  --version       "major.minor" version of target cluster. 

```

Comment:
Can we select tests by labels? OTA qe have some cases which should be executed in a special Prow job. 
For example, I have a case like below:
```
	It("Specified case OTA-12345", Label("OTA-12345"), func() {
		Expect(false).To(BeTrue(), "test failed")
	})
```
Then I want to only run this case.
I tried to run the case like below, it works, but I want to make it more simple:
```
./example-tests run-test --names "[sig-testing] openshift-tests-extension Specified case OTA-12345"
```
Thanks.